src/NativeScript/JSErrors.mm

```diff
@@ -25,12 +25,23 @@ void TNSSetUncaughtErrorHandler(TNSUncaughtErrorHandler handler) {
 
 void reportFatalErrorBeforeShutdown(ExecState* execState, Exception* exception) {
     GlobalObject* globalObject = static_cast<GlobalObject*>(execState->lexicalGlobalObject());
-    globalObject->inspectorController().reportAPIException(execState, exception);
+
+    JSValue uncaughtError = globalObject->get(execState, Identifier::fromString(execState, "__onUncaughtError"));
+    if (uncaughtError.isFunction()) {
```

Comment:
Better replace this with a check that `getCallData` does not return `CallTypeNone` - `isFunction` checks whether a value is a cell that inherits from `JSFunction` or `InternalFunction`, but other cells (proxies, maybe?) can also override `getCallData`.

Also, `isFunction` will return true for ES6 constructor functions, but they aren't `[Call]`-able according to the spec.